test/NuGet.Core.Tests/NuGet.PackageManagement.Test/BuildIntegratedTests.cs

```diff
@@ -1848,14 +1851,2073 @@ public async Task TestPacMan_BuildIntegrated_PreviewUpdatesAsync_WithStrictVersi
             }
         }
 
-        private SourceRepositoryProvider CreateSource(List<SourcePackageDependencyInfo> packages)
+        [Fact]
+        public async Task TestPacMan_PreviewBuildIntegratedProjectsActionsAsync_InstallPackageForAllProjects()
+        {
+            // Arrange
+            var sources = new List<PackageSource>();          
+            var projectDirectories = new List<TestDirectory>();
+
+            try
+            {
+                using (var settingsDirectory = TestDirectory.Create())
+                using (var pathContext = new SimpleTestPathContext())
+                using (var testSolutionManager = new TestSolutionManager())
+                {
+                    // Set up Package Source
+                    var packageA_Version100 = new SimpleTestPackageContext("packageA", "1.0.0");
+
+                    var packageA = packageA_Version100.Identity;
+
+                    await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                        pathContext.PackageSource,
+                        PackageSaveMode.Defaultv3,
+                        packageA_Version100
+                        );
+
+                    sources.Add(new PackageSource(pathContext.PackageSource));
+
+                    var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateSourceRepositoryProvider(sources);
+
+                    var deleteOnRestartManager = new TestDeleteOnRestartManager();
+                    var nuGetPackageManager = new NuGetPackageManager(
+                        sourceRepositoryProvider,
+                        NullSettings.Instance,
+                        testSolutionManager,
+                        deleteOnRestartManager);
+
+                    var testNuGetProjectContext = new TestNuGetProjectContext();
+                    var projectTargetFramework = NuGetFramework.Parse("net46");
+
+                    var configs = new List<string>();
+                    var lockFiles = new List<string>();
+                    var buildIntegratedProjects = new List<TestProjectJsonBuildIntegratedNuGetProject>();
+
+                    // Create projects
+                    for (var i = 0; i < 4; i++)
+                    {
+                        var directory = TestDirectory.Create();
+                        projectDirectories.Add(directory);
+
+                        var config = Path.Combine(directory, "project.json");
+
+                        configs.Add(config);
+
+                        GetBasicConfig(config, "net46");
```

Comment:
lots of spaces between statements makes these methods very long./

Reply:
I reduced number of spaces.